doc/design/high_level_design_document.md

```diff
@@ -0,0 +1,142 @@
+
+8. **Intelligent bots**: The game must include intelligent AI-controlled bots
+
+9. **Interactive environment**: The game environment must include interactable
+     elements such as destructible objects.
+
+10. **End-game statistics**: The game must display detailed statistics at the
+    end of each match.
+
+11. **Monitoring statistics throughout the game**: The game must monitor and
+    display statistics during gameplay.
+
+### Non-functional requirements
+
+1. **Performance**:
+   - The game should target a frame rate of 60 FPS for smooth gameplay.
+   - The system must ensure low-latency communication for real-time interactions.
+
+2. **Scalability**:
+   - The game architecture should handle an increase in the number of players
+     and game sessions efficiently.
+
+3. **Security**:
+   - Player data must be protected against unauthorized access and breaches.
+   - The system must include secure authentication and authorization mechanisms.
+
+4. **Reliability**:
+   - The game should handle network disruptions gracefully, with mechanisms for
+     reconnection and state preservation.
+
+5. **Usability**:
+   - The user interface should be intuitive and easy to navigate.
+   - The game should provide clear feedback to players for their actions.
+
+6. **Maintainability**:
+   - The codebase should be modular and well-documented to facilitate easy
+     maintenance and updates.
+
+### Optional features
+
+1. **Controller support**: The game should optionally support game controllers.
+
+## Architecture overview
+
+### System architecture
+
+Project OMG game architecture is based on a monolithic design, where all
+components are integrated into a single codebase. This approach simplifies
+development and deployment, making it suitable for the project's initial scope.
+The architecture consists of three main components:
+
+- [Client application][client_app_doc]
+- [Game server][game_server_doc]
+- [Database][database_doc]
+
+### Functional requirements coverage by components
+
+| Functional requirement | Client application | Game server | Database |
+|------------------------|--------------------|-------------|----------|
+| Multiplayer support    |                    |      ✓      |          |
+| Top-down 2D view       |         ✓          |             |          |
+| Character lock on camera |       ✓          |             |          |
+| Match-based gameplay   |         ✓          |      ✓      |          |
+| Character creation     |         ✓          |      ✓      |     ✓    |
+| Skill synergy          |         ✓          |      ✓      |          |
+| PvPvE gameplay         |                    |      ✓      |          |
+| Intelligent bots       |                    |      ✓      |          |
+| Interactive environment |        ✓          |             |          |
+| End-game statistics    |         ✓          |      ✓      |     ✓    |
+| Monitoring statistics throughout the game | ✓ |      ✓      |     ✓  |
+
```

Comment:
"PvPvE gameplay", "Intelligent bots" and "Interactive environment" should both be included in client application and game server

Reply:
This part was indeed hard to put it on a table. I edited it according to your suggestions. But in practice, the things on the table are subject-to-change. It might be hard to fit the requirements to clear components, or the other way around.

done.